discovery/discovery.go

```diff
@@ -55,12 +59,65 @@ func (o *orchestratorPool) GetInfos() []common.OrchestratorLocalInfo {
 	return o.infos
 }
 
+func ResolveURL(urlToResolve *url.URL) []*url.URL {
+	resolvedUrls := make([]*url.URL, 0, 0)
+	// check if there's a port in host, for ipv6 it's after ]
```

Comment:
Isn't `urlToResolve` already parsed to `host` and `port`. I think that `urlToResolve.Host` should contain host (without port) and `urlToResolve.Port()` should contain the port.

If that's not the case, I guess we should just use `url.ParseRequestURI()` instead of this custom logic below.

Reply:
Thanks for suggestion, replaced with library functions.